src/components/SearchFilterList/SearchFilter.js

```diff
@@ -0,0 +1,112 @@
+import { useState, useMemo } from 'react'
+import { Box, CheckBox, Heading, Text } from 'grommet'
+import { Button as sharedButton } from 'components/shared/Button'
+import { SearchBox } from 'components/shared/SearchBox'
+import { formatString } from 'helpers/formatString'
+import { scrollToId } from 'helpers/scrollToId'
+import styled, { css } from 'styled-components'
+
+const ToggleButton = styled(sharedButton)`
+  border-bottom: 1px solid transparent;
+  ${({ theme }) => css`
+    &:hover {
+      border-bottom: 1px solid ${theme.global.colors.brand};
+    }
+  `}
+`
+
+export const SearchFilter = ({ filterGroup, label }) => {
+  const options = useMemo(() => {
+    return Object.entries(filterGroup)
+  }, [filterGroup])
+  const [filteredResult, setfilteredResult] = useState(options)
+  const [open, setOpen] = useState(false)
+  const [userInput, setUserInput] = useState('')
+  const maxCount = 5
+  const filterLength = options.length
+
+  const filterOptions = (val) => {
+    setUserInput(val)
+
+    if (val.trim() !== '') {
+      setfilteredResult(() =>
+        options.filter((option) =>
+          formatString(option[0]).toLowerCase().startsWith(val.toLowerCase())
+        )
+      )
+    } else {
+      setfilteredResult(options)
+    }
+  }
+
+  const getOptionsToRender = () =>
+    open ? filteredResult : filteredResult.slice(0, maxCount)
+
+  return (
+    <>
+      <Heading level={4} margin={{ bottom: 'xsmall' }} id={label.toLowerCase()}>
+        {label}
+      </Heading>
+
+      {filterLength > maxCount && (
+        <SearchBox
+          pad={{ bottom: 'xsmall' }}
+          placeholder={`Filter ${label}`}
+          value={userInput}
+          size="small"
+          changeHandler={(e) => filterOptions(e.target.value)}
+        />
+      )}
+
+      <Box
+        margin={{ top: 'xsmall' }}
+        animation={open ? { type: 'fadeIn', duration: 1000 } : {}}
+      >
+        {getOptionsToRender().map((option, i, arr) => (
+          <Box
+            key={option[0]}
+            margin={{ bottom: i !== arr.length - 1 ? 'xsmall' : '0' }}
```

Comment:
I think an `isLast` helper would help here ex: isLast(arr, i, '0', 'xsmall')

Reply:
Added a helper `isLastIndex(arr, index)` to determine whether the given index is the last element of an array, thank you!

**NOTE:** To keep the name of the helper semantic, it simply checks the position of an index